Allow the usage of UUID's to find Offline Users

FixedDev opened the pull request:
### **First of all, how came I to the conclusion that this change was needed?**
I was using the plugin as a server owner and due to administrative purposes I needed to teleport into the home of an offline User, but essentials didn't resolve the UUID correctly, so I tried to use directly the UUID to teleport into that user's home but that wasn't allowed by essentials.
### **What are the changes?**
There's only one change, when the EssentialsCommand#getPlayer(Server, User, String, boolean, boolean) method is called with a search term that's a UUID and the user is offline(exPlayer == null) we try to parse the searchTerm as a UUID first, if the parsing fails with an IllegalArgumentException then the searchTerm is used as a String.

FixedDev committed:
Update EssentialsCommand.java

Allow the usage of UUID's to get offline users

FixedDev committed:
Update EssentialsCommand.java

Set the term to use in an object variable, then use an instanceof to call the correct method

Octol1ttle commented on Essentials/src/com/earth2me/essentials/commands/EssentialsCommand.java:
```diff
@@ -145,7 +145,18 @@ private User getPlayer(final Server server, final User sourceUser, final String
+            Object searchId;
+            try {
+                searchId = UUID.fromString(searchTerm);
+            }catch(final IllegalArgumentException ex) {
```
No space after `}`

FixedDev replied:
Done

FixedDev committed:
Update EssentialsCommand.java

Added a missing space in line 151 before te catch

mdcfe commented on Essentials/src/com/earth2me/essentials/commands/EssentialsCommand.java:
```diff
@@ -145,7 +145,18 @@ private User getPlayer(final Server server, final User sourceUser, final String
             user = ess.getUser(exPlayer);
         } else {
-            user = ess.getUser(searchTerm);
+            Object searchId;
+            try {
+                searchId = UUID.fromString(searchTerm);
+            } catch(final IllegalArgumentException ex) {
+                searchId = searchTerm;
+            }
+            
+            if (searchId instanceof String) {
+                user = ess.getUser((String) searchId);
+            } else { 
+                user = ess.getUser((UUID) searchId);
+            }
```
This... just... no. Please don't abuse types like this.

FixedDev replied:
done

mdcfe commented:
> I was using the plugin as a server owner and due to administrative purposes I needed to teleport into the home of an offline User, but essentials didn't resolve the UUID correctly, so I tried to use directly the UUID to teleport into that user's home but that wasn't allowed by essentials.

From this description, you are attempting to fix a bug. If this is the case, please [open a bug report](https://github.com/EssentialsX/Essentials/issues/new/choose) with all the details required. If there's an incorrect behaviour that we need to fix, we will then either link this PR to your or implement a suitable fix.

FixedDev committed:
Update EssentialsCommand.java

Instead of abusing of "instanceof" and casting, use a mutable variable and set the final user variable after that.

FixedDev commented:
> > I was using the plugin as a server owner and due to administrative purposes I needed to teleport into the home of an offline User, but essentials didn't resolve the UUID correctly, so I tried to use directly the UUID to teleport into that user's home but that wasn't allowed by essentials.
> 
> From this description, you are attempting to fix a bug. If this is the case, please [open a bug report](https://github.com/EssentialsX/Essentials/issues/new/choose) with all the details required. If there's an incorrect behaviour that we need to fix, we will then either link this PR to your or implement a suitable fix.

I think that it wasn't a bug actually, just a behavior that wasn't implemented at the moment, that means that the closest thing to this would be a pull request

mdcfe commented:
> 
> 
> > > I was using the plugin as a server owner and due to administrative purposes I needed to teleport into the home of an offline User, but essentials didn't resolve the UUID correctly, so I tried to use directly the UUID to teleport into that user's home but that wasn't allowed by essentials.
> > 
> > 
> > From this description, you are attempting to fix a bug. If this is the case, please [open a bug report](https://github.com/EssentialsX/Essentials/issues/new/choose) with all the details required. If there's an incorrect behaviour that we need to fix, we will then either link this PR to your or implement a suitable fix.
> 
> I think that it wasn't a bug actually, just a behavior that wasn't implemented at the moment

> I needed to teleport into the home of an offline User, but essentials didn't resolve the UUID correctly

This is a bug. Please report it properly.

FixedDev commented:
> > > > I was using the plugin as a server owner and due to administrative purposes I needed to teleport into the home of an offline User, but essentials didn't resolve the UUID correctly, so I tried to use directly the UUID to teleport into that user's home but that wasn't allowed by essentials.
> > > 
> > > 
> > > From this description, you are attempting to fix a bug. If this is the case, please [open a bug report](https://github.com/EssentialsX/Essentials/issues/new/choose) with all the details required. If there's an incorrect behaviour that we need to fix, we will then either link this PR to your or implement a suitable fix.
> > 
> > 
> > I think that it wasn't a bug actually, just a behavior that wasn't implemented at the moment
> 
> > I needed to teleport into the home of an offline User, but essentials didn't resolve the UUID correctly
> 
> This is a bug. Please report it properly.

That part may be a bug, but this pull request isn't about that bug, is adding a feature that in some cases can be needed

JRoy commented:
Closed due to being stale
